### PR TITLE
feat(tools): add bazelisk

### PR DESCRIPTION
**Pull request by aschleifer**
closes https://github.com/containerbase/base/issues/20

This adds `bazelisk` as possible tool to the containerbase. That way it can be used in renovate.

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
can you please don't like flux, so I don't need to convert it after merge?

https://github.com/containerbase/base/blob/main/src/cli/tools/flux/index.ts

**Comment by aschleifer** [comment]
> can you please don't like flux, so I don't need to convert it after merge?
> 
> https://github.com/containerbase/base/blob/main/src/cli/tools/flux/index.ts

Hello, do I understand you correctly that you want me to do it in typescript? If so and flux is the example then I can do it next week for sure.

**Comment by viceice** [comment]
yes, thanks 🙏

**Comment by aschleifer** [comment]
Hello @viceice ,

I rewrote the tool in typescript.

As an improvement for the future it might help to have some guide explaining how to to do it. I found the other file changes (not in the `src/cli/tools/bazelisk` folder) out by grepping for "flux" and reading the code. 

Also I didn't find any documentation stating what the "prepare" stage is suppose to be used for.

**Review by viceice — COMMENTED** [review]
please add a test install 

https://github.com/containerbase/base/blob/cddf64e46c78e129fa5a9942064a9f5ebdf5fa75/test/Dockerfile.jammy#L33

https://github.com/containerbase/base/blob/cddf64e46c78e129fa5a9942064a9f5ebdf5fa75/test/latest/Dockerfile.arm64#L25

**Comment by viceice** [comment]
please don't force push

File: src/cli/tools/bazelisk/index.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    });
+  }
+
+  override async link(version: string): Promise<void> {
+    const src = join(this.pathSvc.versionedToolPath(this.name, version), 'bin');
+
+    await this.shellwrapper({
+      srcDir: src,
+    });
+    await fs.symlink('bazelisk', join(this.pathSvc.binDir, 'bazel'));
```
[the remark this answers is not in the document]
Comment (aschleifer) [reply src/cli/tools/bazelisk/index.ts:56]: why does this look wrong? `bazel` should be using the `bazelisk` executable.

**Comment by aschleifer** [comment]
> please add a test install
> 
> https://github.com/containerbase/base/blob/cddf64e46c78e129fa5a9942064a9f5ebdf5fa75/test/Dockerfile.jammy#L33
> 
> https://github.com/containerbase/base/blob/cddf64e46c78e129fa5a9942064a9f5ebdf5fa75/test/latest/Dockerfile.arm64#L25

I added some tests. Let me know if something is missing.

File: src/cli/tools/bazelisk/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (viceice) [reply src/cli/tools/bazelisk/index.ts:56]: sure, but the first argument of node fs symlink needs to be the full path or relative to cwd (which is unknown and can be arbitrary at runtime)

[the remark this answers is not in the document]
Comment (aschleifer) [reply src/cli/tools/bazelisk/index.ts]: > sure, but the first argument of node fs symlink needs to be the full path or relative to cwd (which is unknown and can be arbitrary at runtime)

Good point. I changed the symlink to be absolute now.

File: test/latest/Dockerfile.arm64 [hunk 2]
```diff
@@ -16,6 +16,15 @@ COPY src/ /
...
 RUN install-containerbase
 
+#--------------------------------------
+# Image: bazelisk
+#--------------------------------------
+FROM base as test-bazelisk
+
+
+# renovate: datasource=github-tags packageName=bazelbuild/bazelisk
+RUN install-tool bazelisk v1.18.0
```
Comment (viceice) [inline test/latest/Dockerfile.arm64:26]:
```suggestion
RUN install-tool bazelisk v1.18.0
RUN bazel --version
```

test symlink 🙃

File: src/cli/tools/bazelisk/index.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    @inject(EnvService) envSvc: EnvService,
+    @inject(PathService) pathSvc: PathService,
+    @inject(HttpService) private http: HttpService,
+  ) {
+    super(pathSvc, envSvc);
+  }
+
+  override async install(version: string): Promise<void> {
+    const baseurl = `https://github.com/bazelbuild/bazelisk/releases/download/v${version}/`;
+    const filename = `bazelisk-linux-${this.arch}`;
```
Comment (viceice) [inline src/cli/tools/bazelisk/index.ts:26]:
```suggestion
    const filename = `bazelisk-linux-${this.envSvc.arch}`;
```

File: src/cli/tools/bazelisk/index.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import { EnvService, HttpService, PathService } from '../../services';
+
+@injectable()
+export class InstallBazeliskService extends InstallToolBaseService {
+  readonly name = 'bazelisk';
+
+  private get arch(): string {
+    return this.envSvc.arch;
+  }
+
```
Comment (viceice) [inline src/cli/tools/bazelisk/index.ts:15]:
```suggestion
```
We don't need this simple property

File: test/latest/Dockerfile.arm64 [hunk 5]
```diff
@@ -16,6 +16,16 @@ COPY src/ /
...
 
+#--------------------------------------
+# Image: bazelisk
+#--------------------------------------
+FROM base as test-bazelisk
+
+
+# renovate: datasource=github-releases packageName=bazelbuild/bazelisk
+RUN install-tool bazelisk v1.18.0
+RUN bazel --version
```
Comment (viceice) [inline test/latest/Dockerfile.arm64:27]:
```suggestion
```

**Review by viceice — APPROVED** [review]

**Pull request merged**